tests/foreman/cli/test_registration.py

```diff
@@ -16,8 +20,53 @@
 
 from robottelo.config import settings
 
+pytestmark = pytest.mark.tier1
+
+
+@pytest.mark.e2e
+@pytest.mark.no_containers
+@pytest.mark.rhel_ver_match('[^6].*')
+def test_host_registration_end_to_end(
+    module_org,
+    module_location,
+    module_ak_with_synced_repo,
+    module_target_sat,
+    module_capsule_configured,
+    rhel_contenthost,
+):
+    """Verify content host registration with global registration
+
+    :id: b6cd60ba-8069-11ed-ac61-83315855d126
+
+    :steps:
+        1. Register host with global registration template to Satellite and Capsule
+
+    :expectedresults: Host registered successfully
+    """
+    result = rhel_contenthost.register(
+        module_org, module_location, module_ak_with_synced_repo.name, satellite=module_target_sat
+    )
+    assert result.status == 0, f'Failed to register host: {result.stderr}'
+
+    # Update module_capsule_configured to include module_org/module_location
+    module_target_sat.cli.Capsule.update(
+        {
+            'name': module_capsule_configured.hostname,
+            'organization-ids': module_org.id,
+            'location-ids': module_location.id,
+        }
+    )
+    result = rhel_contenthost.register(
+        module_org,
+        module_location,
+        module_ak_with_synced_repo.name,
+        target=module_capsule_configured,
```

Comment:
Non blocking, can we use the `capsule` keyword itself instead of `target`

Reply:
@adarshdubey-star Will the target be always a Satellite Capsule if it is used?

Reply:
Yes, it will be satellite as default, if smart_proxy option is used then capsule

Reply:
That is actually a good idea @adarshdubey-star. 
@Gauravtalreja1 Can we rename `target` to `capsule` and remove the `satellite` parameter?
at the end of the function, instead of 

```python
cmd = satellite.cli.HostRegistration.generate_command(options)
```

you would use

```python
cmd = capsule.satellite.cli.HostRegistration.generate_command(options)
```
WDYT about it?

Reply:
capsule class doesn't contain any satellite property which needs to be implemented for this, 
for capsule we need to run commands on satellite where its integrated, I couldn't find any simple way to get this hostname for sat, found only way with foreman-proxy service from which it seem little tricky to get sat hostname, so I'd prefer to skip this, as its not a blocker and implement this later :)